packages/simple-hub/src/message/on-message.ts

```diff
@@ -14,8 +15,8 @@ export function onIncomingMessage(
     snapshotKey: string;
     message: Either<Error, Message>;
   }>,
-  onNext?: () => void,
-  onComplete?: () => void
+  onNext: () => void = noop,
+  onComplete: () => void = noop
```

Comment:
Outside of unit testing, the subscriber should never see completion of the observable. If the subscriber does see a completion of the stream, something has gone wrong.
```suggestion
  onComplete?: () => void
```